automation/scripts/check_fully_qualified_imports.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# Copyright 2023 Iguazio
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import pathlib
+import re
+import sys
+
+TARGET_MODULES = {"mlrun", "mlrun_pipelines"}
+
+DISALLOWED_PATTERN = re.compile(r"^\s*from\s+(mlrun|mlrun_pipelines)\s+import")
+
+
+def check_disallowed_imports(file_path: pathlib.Path) -> list[str]:
+    """
+    Check for disallowed imports in a file and return a list of violations.
+    """
+    violations = []
+    with open(file_path) as file:
+        for line_num, line in enumerate(file, start=1):
+            if DISALLOWED_PATTERN.match(line.strip()):
```

Comment:
I think you can skip lines that don't start with or have the word "from", because you don't really need to match them.

Reply:
I think this is probably premature optimization, regex is fast enough to not process the entire line if the beginning doesnt match